li-apache-kafka-clients/src/main/java/com/linkedin/kafka/clients/consumer/FederatedSubscriptionState.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2019 LinkedIn Corp. Licensed under the BSD 2-Clause License (the "License").  See License in the project root for license information.
+ */
+
+package com.linkedin.kafka.clients.consumer;
+
+import java.util.Collections;
+import java.util.Set;
+import java.util.regex.Pattern;
+
+import org.apache.kafka.common.TopicPartition;
+
+// Current subscription/assignment state at the federated level
+class FederatedSubscriptionState {
+  public enum SubscriptionType {
+    NONE, AUTO_TOPICS, AUTO_PATTERN, USER_ASSIGNED
```

Comment:
should these be named closer to the kafka terminology: manual_assignment, subscription and wildcard_subscription ?

Reply:
These are what vanilla Kafka currently uses for its SubscriptionState.

Reply:
if those are the vanilla kafka terms, than stick to them.

Reply:
Done